pkg/ratelimiter/rate_limiter.go

```diff
@@ -10,15 +10,15 @@ import (
 )
 
 const (
-	ALLOW_LISTED_RATE_PER_MINUTE = uint16(10)
-	ALLOW_LISTED_MAX_TOKENS      = uint16(10000)
-	REGULAR_RATE_PER_MINUTE      = uint16(1)
-	REGULAR_MAX_TOKENS           = uint16(100)
-	MAX_UINT_16                  = 65535
+	PRIORITY_RATE_PER_MINUTE = uint16(10)
+	PRIORITY_MAX_TOKENS      = uint16(10000)
+	REGULAR_RATE_PER_MINUTE  = uint16(1)
```

Comment:
We should adjust these rates to match the current WAF limits as best we can. Current WAF limits are 200 publishes + 2k queries/minute.

So, if we adjusted the cost of publishing to 10 per publish it would be 4000/minute with a `REGULAR_MAX_TOKENS` of 20k. 

That feels high, since a spammer could just use their entire allocation on publishing and none on querying and get 2000/5 minutes/node. Makes me wonder if we actually want to track the limits separately depending on request type.

Reply:
Agreed, I raised the same questions in the Notes. I also wonder if stream requests should be limited differently given their distinct nature. I can definitely look into separating the limits, I agree to address different requests types with single limit will have weird edge cases.

Reply:
If we do different limits for different request types, do we want different priority limits too?

Reply:
I'd suggest a blanket 5X increase for all rate limits if you have priority status. That feels high enough to be worth doing, but not so high they are going to risk network stability.

My vote is for separate limits per RPC. Especially now that this repo is open source, people will figure out how to game the combined limits very quickly.

Reply:
OK, so let me spell it out in details. Per request limits ( trying to match the edge limits ) :

| RPC Type | rate / max |
| ---------- | --------------|
| Query, BatchQuery(per query) | 2000/10000 |
| Publish (per envelope) | 200 / 1000 |
| Subscribe/Subscribe2/SubscribeAll | 2000 / 10000 |

Priority limits will be fixed multiple of the regular limits. We can make the multiplier configurable and default to 5.

I'd be tempted to dial the streaming calls down a fair bit, but DD shows that we are averaging about twice as many subscribe call (in prod) than query calls over the past week.

<img width="1208" alt="Screenshot 2023-08-25 at 14 01 41" src="https://github.com/xmtp/xmtp-node-go/assets/871693/2fb2a638-cefe-4263-8e8e-45831bf63f14">

Reply:
I should note that limit `2000/10000` allows a spike of 18000 in the first 5 minutes assuming full token budget, but it can't be sustained over longer time, since you have 0 tokens at the end of it and have to stick with the 2000 refill rate per minute.

Reply:
I think that sounds reasonable as a starting place. I'd rather start with these limits being a little generous, and then dial back, rather than breaking any applications on release.

Reply:
I'm OK with leaving the subscribe calls at the same limits as Query. While you're right that the worst case Subscribe call is much more expensive than a Query, we'd have to gather some data to really know what is more expensive. A Query call always hits the database, and a Subscribe call never does. I'd bet most subscriptions don't receive a single message in their lifetime, so it's really just the overhead of handling the connection and setting up the broadcaster. But a subscription to a popular topic would be quite expensive.